Incorrect codegen for VCC compiler

Nim compiler's codegen produces incorrect code for VCC.

### Example
```nim
let a: array[0, byte] = []

proc someProc(a: openarray[byte]): int =
  len(a)

when isMainModule:
  echo someProc(a)
```

### Current Output
```
nim c --cc:vcc emptyarray.nim

Hint: used config file 'C:\Projects\nim-beacon-chain\vendor\nimbus-build-system\vendor\Nim\config\nim.cfg' [Conf]
Hint: used config file 'C:\Projects\nim-beacon-chain\vendor\nimbus-build-system\vendor\Nim\config\config.nims' [Conf]
Hint: system [Processing]
Hint: widestrs [Processing]
Hint: io [Processing]
Hint: emptyarray [Processing]
CC: emptyarray.nim
@memptyarray.nim.c
nimcache\emptyarray_d\@memptyarray.nim.c(62): error C2059: syntax error: '}'
```

### Expected Output
```
0
```

### Additional Information
MinGW compiler works fine with produced c code.

```
Nim Compiler Version 1.2.2 [Windows: amd64]
Compiled at 2020-06-18
Copyright (c) 2006-2020 by Andreas Rumpf

git hash: be34b5abe23e027b71a3a9da67037aac7afea6bc
active boot switches: -d:release
```

--- a/compiler/ccgexprs.nim
+++ b/compiler/ccgexprs.nim
@@ -3331,6 +3331,8 @@ proc genConstObjConstr(p: BProc; n: PNode; isConst: bool; result: var Rope) =
 
 proc genConstSimpleList(p: BProc, n: PNode; isConst: bool; result: var Rope) =
   result.add "{"
+  if p.vccAndC and n.len == 0 and n.typ.kind == tyArray:
+    getDefaultValue(p, n.typ[1], n.info, result)
   for i in 0..<n.len:
     let it = n[i]
     if i > 0: result.add ",\n"

--- a/tests/array/t15117.nim
+++ b/tests/array/t15117.nim
@@ -1,0 +1,27 @@
+discard """
+  matrix: "--cc:vcc"
+  disabled: "linux"
+  disabled: "bsd"
+  disabled: "osx"
+  disabled: "unix"
+  disabled: "posix"
+"""
+{.experimental: "views".}
+
+let a: array[0, byte] = []
+discard a
+
+type B = object
+  a:int
+let b: array[0, B] = []
+let c: array[0, ptr B] = []
+let d: array[0, ref B] = []
+discard b
+discard c
+discard d
+
+discard default(array[0, B])
+
+type
+  View1 = openArray[byte]
+discard default(View1)

